content/React Query and management of server state - Nicolás Santos.md

```diff
@@ -0,0 +1,59 @@
+
+Before jumping off to the code and seeing some examples I'd like to introduce how React Query models the server state.
+![React query async states](https://i.ibb.co/4M4bkQ8/Screen-Shot-2020-05-14-at-14-23-23.png)
+In the image above, we can see the different states for the server state information, let's explain what is the meaning of each state:
+- Fetching: This is the initial state and occurs when the data is fetching from an outsource, typically the backend.
+- Fresh: On this state, the information is the same on both sides, this is the desired state of our application because it implies that we don't need to re-fetch the information. On the other hand, this state lasts a short time because in most cases when you're fetching information, this information immediately is potentially out of date.
+- Stale: Represents the out of date data that the app is currently using. This happens because the frontend has old information and needs to re-fetch it from the backend, or the backend is not updated because the frontend didn't send the new information to the backend yet. This state is particularly important as long as we want to be in the fresh state again.
+- Inactive: React Query has a garbage collector for managing cache on the browser, this state in some manner indicates to React query that if the data is not being used in the application, it will potentially be deleted after a while. This is a great feature of React Query, because we want to persist the recent information to not fetching data all the time and improve the speed/UX of our interfaces.
+- Deleted: This happens when the data was inactive for a certain period of time and it's deleted from the cache. This timeout can be configurable locally for each query or globally.
+
+With this approach, React Query handles the asynchronous information of our application clearly, allowing us to scale and maintain our code in a better way.
+
+## Introduction to React query API
+Although React Query has only two hooks, it is highly configurable in all aspects, from the retry delay time of the queries to the set maximum time of inactive data in the cache.
+But let's start with the most basic API that is well documented in [their repository](https://github.com/tannerlinsley/react-query)
+
+https://gist.github.com/nsantos16/7f71e979ed3c884755906c9dbdee1436
+
+This hook provided us the status of the fetch (loading, error, or success), and data and error if they are defined. Up to this point is a pretty normal hook for fetching information, but as we see the query is related to a key (in this case `movies`), this is a unique-global key that is used for associate the query information between the components, in this way we can re-use this hook to use the same information in anywhere of the DOM tree. If you follow the classic redux action-reducer cycle for fetching data, **this hooks it will save up a lot of repetitive code**.
```

Comment:
And here `**this hook will save up a lot of repetitive code**` singular and without the `it`